### ISSUE TITLE
[BUG] Textures missing when export OBJ

### ISSUE THREAD

**Issue by YukihiraDeve** [jmc2obj/j-mc-2-obj#251]
Hello,
When I using jMc2Obj-124.jar on my Debian server, with this command line 

``` java -jar jMc2Obj-124.jar /servers/world/ ```

I have this result.
![Capture d’écran 2024-06-01 à 18 24 54](https://github.com/jmc2obj/j-mc-2-obj/assets/82152262/dd7bafab-e795-48dd-9639-247e0084ffd0)

the MTL files have size 0ko.

In the console, we have a lot of error. Lot of texture block missing and available. 

I try to add minecraft.jar link, but a args not be unavaiblable

this is a log when I export my map : 
```
Exporting world /servers/tset/world
Processing chunks...
[...Couldn't find blockstate minecraft:stone in any resource pack!
Couldn't get blockstate to export minecraft:stone
Couldn't find blockstate minecraft:andesite in any resource pack!
Couldn't get blockstate to export minecraft:andesite
Couldn't find blockstate minecraft:coal_ore in any resource pack!
Couldn't get blockstate to export minecraft:coal_ore
Couldn't find blockstate minecraft:diorite in any resource pack!
Couldn't get blockstate to export minecraft:diorite
Couldn't find blockstate minecraft:dirt in any resource pack!
Couldn't get blockstate to export minecraft:dirt
Couldn't find blockstate minecraft:granite in any resource pack!
Couldn't get blockstate to export minecraft:granite
Couldn't find blockstate minecraft:gravel in any resource pack!
Couldn't get blockstate to export minecraft:gravel
Couldn't find blockstate minecraft:grass_block in any resource pack!
Couldn't get blockstate to export minecraft:grass_block
Couldn't find blockstate minecraft:copper_ore in any resource pack!
Couldn't get blockstate to export minecraft:copper_ore
Couldn't find blockstate minecraft:iron_ore in any resource pack!
Couldn't get blockstate to export minecraft:iron_ore
Couldn't find blockstate minecraft:short_grass in any resource pack!
Couldn't get blockstate to export minecraft:short_grass
Couldn't find blockstate minecraft:deepslate in any resource pack!
Couldn't get blockstate to export minecraft:deepslate
Couldn't find blockstate minecraft:birch_leaves in any resource pack!
Couldn't get blockstate to export minecraft:birch_leaves
Couldn't find blockstate minecraft:deepslate_lapis_ore in any resource pack!
Couldn't get blockstate to export minecraft:deepslate_lapis_ore
Couldn't find blockstate minecraft:birch_log in any resource pack!
Couldn't get blockstate to export minecraft:birch_log
Couldn't find blockstate minecraft:deepslate_iron_ore in any resource pack!
Couldn't get blockstate to export minecraft:deepslate_iron_ore
Couldn't find blockstate minecraft:glow_lichen in any resource pack!
Couldn't get blockstate to export minecraft:glow_lichen
Couldn't find blockstate minecraft:deepslate_redstone_ore in any resource pack!
Couldn't get blockstate to export minecraft:deepslate_redstone_ore
Couldn't find blockstate minecraft:redstone_ore in any resource pack!
Couldn't get blockstate to export minecraft:redstone_ore
Couldn't find blockstate minecraft:lapis_ore in any resource pack!
Couldn't get blockstate to export minecraft:lapis_ore
Couldn't find blockstate minecraft:gold_ore in any resource pack!
Couldn't get blockstate to export minecraft:gold_ore
Couldn't find blockstate minecraft:deepslate_copper_ore in any resource pack!
Couldn't get blockstate to export minecraft:deepslate_copper_ore
Couldn't find blockstate minecraft:tuff in any resource pack!
Couldn't get blockstate to export minecraft:tuff
Couldn't find blockstate minecraft:magma_block in any resource pack!
Couldn't get blockstate to export minecraft:magma_block
...Couldn't find blockstate minecraft:diamond_ore in any resource pack!
Couldn't get blockstate to export minecraft:diamond_ore
...............................................Couldn't find blockstate minecraft:deepslate_coal_ore in any resource pack!
Couldn't get blockstate to export minecraft:deepslate_coal_ore
.....................Reading Chunks:1.277571435
....]
Writing File:0.002862611
OBJ Export Time:1.281060707
Saved model to /home/val/WebServerController/Backend/export/./minecraft.obj
Sorting OBJ file...
[..............................................................................]
Writing materials to /home/val/WebServerController/Backend/export/./minecraft.mtl...
Error writing material definition minecraft:block/lava_flow
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/lava_flow.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.registry.TextureEntry.hasAlpha(TextureEntry.java:87)
	at org.jmc.Materials.writeMTLFile(Materials.java:57)
	at org.jmc.ObjExporter.export(ObjExporter.java:375)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
[...................Error writing material definition minecraft:block/water_still
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/water_still.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.registry.TextureEntry.hasAlpha(TextureEntry.java:87)
	at org.jmc.Materials.writeMTLFile(Materials.java:57)
	at org.jmc.ObjExporter.export(ObjExporter.java:375)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
....................Error writing material definition minecraft:block/lava_still
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/lava_still.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.registry.TextureEntry.hasAlpha(TextureEntry.java:87)
	at org.jmc.Materials.writeMTLFile(Materials.java:57)
	at org.jmc.ObjExporter.export(ObjExporter.java:375)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
...................Error writing material definition minecraft:block/water_flow
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/water_flow.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.registry.TextureEntry.hasAlpha(TextureEntry.java:87)
	at org.jmc.Materials.writeMTLFile(Materials.java:57)
	at org.jmc.ObjExporter.export(ObjExporter.java:375)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
....................]
Exporting textures...
Couldn't export texture minecraft:block/lava_flow
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/lava_flow.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.TextureExporter.exportTextures(TextureExporter.java:159)
	at org.jmc.ObjExporter.export(ObjExporter.java:386)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
[...................Couldn't export texture minecraft:block/water_still
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/water_still.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.TextureExporter.exportTextures(TextureExporter.java:159)
	at org.jmc.ObjExporter.export(ObjExporter.java:386)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
....................Couldn't export texture minecraft:block/lava_still
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/lava_still.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.TextureExporter.exportTextures(TextureExporter.java:159)
	at org.jmc.ObjExporter.export(ObjExporter.java:386)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
...................Couldn't export texture minecraft:block/water_flow
java.io.FileNotFoundException: Couldn't find assets/minecraft/textures/block/water_flow.png in current resource packs
	at org.jmc.util.ResourcePackIO.loadResource(ResourcePackIO.java:71)
	at org.jmc.util.ResourcePackIO.loadResourceAsStream(ResourcePackIO.java:43)
	at org.jmc.util.ResourcePackIO.loadImage(ResourcePackIO.java:26)
	at org.jmc.registry.TextureEntry.getImage(TextureEntry.java:109)
	at org.jmc.TextureExporter.exportTextures(TextureExporter.java:159)
	at org.jmc.ObjExporter.export(ObjExporter.java:386)
	at org.jmc.Main.runConsole(Main.java:109)
	at org.jmc.Main.main(Main.java:51)
....................]
Export Time:1.321434358
Done! ```

I've checked the administrator and writing rights, and the version is indeed the right one.

If you have an idea thank you a lot :)

**Comment by Viottery** [comment]
hi, I would like to know how you dealt with this, cause I met the same problem. Thanks!

**Comment by YukihiraDeve** [comment]
Hi,
I used it this line with my terminal script.sh : 

`java -jar "$jmc2objPath" --resource-pack="$MinecraftJarPath" "$worldPath" --area="$minSpawnX,$minSpawnZ,$((spawnX+60)),$((spawnZ+60))" --offset=center --object-per-mat --height="$min,$maxY"
`

and for $MinecraftJarPath, I used this : 
MINECRAFT_JAR="https://piston-data.mojang.com/v1/objects/05b6f1c6b46a29d6ea82b4e0d42190e42402030f/client.jar"

The problem is that the Jmc tool does not have the default textures. By default it uses .minecraft. Except that on a Linux server it is not necessarily available, so I downloaded a client.jar and it was good!

> hi, I would like to know how you dealt with this, cause I met the same problem. Thanks!
